xoinvader/background.py

```diff
@@ -0,0 +1,261 @@
+                             "any chunks".format(filename))
+        return chunks
+
+
+class Background(Renderable):
+    """Class for storing and managing level background.
+    Manages background rendering and changing in time.
+
+    :param loop: flag if current chunk should loop after it had ended
+    :type loop: boolean
+
+    :param loop_all: flag if whole background should repeat from beginning
+    :type loop: boolean
+
+    :param speed: speed of background advance. Can be negative to represent
+    moving backwards
+    :type speed: integer
+
+    :param chunks: list of background chunks
+    :type chunks: list
+
+    :param background: list of background lines. You may change it at any time.
+    This list directly converts into Surface which then goes to the renderer
+    :type list:
+
+    """
+    render_priority = -1000
+
+    def __init__(self, filename=None):
+        self.loop = False
+        self.loop_all = False
+        self.speed = 0
+        self.chunks = []
+        self.background = []
+
+        self._w = Settings.layout.field.edge.x
+        self._h = Settings.layout.field.edge.y
+        self._background_surface = None
+        self._current_chunk = None
+        self._current_chunk_num = 0  # position of chunk in chunk list
+        self._chunk_line = 0  # position in current chunk
+        self._ticks_since_last_update = 0
+
+        if filename:
+            self.load_file(filename)
+
+    def clear(self):
+        """Clear all background.
+        Doesn't prevent updating, if speed is not 0."""
+        self.background = [' ' * self._w] * self._h
+
+    def load_file(self, filename):
+        self.chunks = load_chunks(filename, self._w)
+
+    def _fill(self):
+        """Fills the whole background"""
+        self.background = []
+        for i in range(self._h):
+            self.background.append(self._advance_chunk(1))
+
+
+    def start(self, filled=False):
+        """Start background from the beginning. If `filled` parameter is
+        provided, fills the background starting with first chunk.
+
+        :param filled: if true, perform initial background fill
+        :type filled: boolean"""
+        self._ticks_since_last_update = 0
+        self._current_chunk_num = 0
+        self._current_chunk = self.chunks[0]
+        self._chunk_line = 0
+        if filled:
+            self._fill()
+        else:
+            self.clear()
+        self.update_surface()
+
+    def _advance_chunk(self, advance):
+        """Return current chunk line and update current chunk's position,
+        or current chunk if there's no lines left in it.
+
+        :return: current line in chunk sequence
+        :rtype: string
+        """
+        if not self._current_chunk:
+            return ' ' * self._w
+
+        # start checks
+        if (self._chunk_line < self._current_chunk.length and
+                self._chunk_line >= 0):
+            l = self._current_chunk[self._chunk_line]
```

Comment:
Sorry, l is not very good name, I thought that you return 1. And linter will be confused too by such short name.

If you very very very need to use name that will be marked by Pylint as invalid - you can add comment `# pylint: disable=invalid-name`.